vllm/sampling_params.py

```diff
@@ -305,12 +291,12 @@ def update_from_generation_config(
                 # purposes.
                 eos_ids.discard(model_eos_token_id)
             if eos_ids:
-                self.all_stop_token_ids.update(eos_ids)
+                self._all_stop_token_ids.update(eos_ids)
                 if not self.ignore_eos:
                     eos_ids.update(self.stop_token_ids)
                     self.stop_token_ids = list(eos_ids)
 
-    @cached_property
+    @property
```

Comment:
Does that mean the sampling type will change on th fly?

Reply:
yes. I assume it is pretty cheap though?